### PR TITLE
User Management API

### PR DESCRIPTION
**Pull request by KurtXiaoZ**
This pr includes all of the user management APIs except for conditional readUsers.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -7,6 +7,8 @@ require('dotenv').config();
 const app = express();
 const port = process.env.PORT || 8080;
 
+global.__basedir = __dirname + "/..";
```
Comment (Vivian-liyw) [inline index.js:10]: Would you happen to know where we are using this?

File: controllers/userController.js [hunk 2]
```diff
@@ -1,90 +1,348 @@
...
-    if (data[0].affectedRows) {
-      message = 'user deleted successfully';
-      res.status(200).send({message: message, data: data[0]});
+    if (data[0].length === 0) {
+      console.log(data[0])
+      return res.send({status: STATUS_CODE.INVALID_EMAIL});
+    } else {
+      //data[0][0] contains the actual json object that gets returned by mysql
+      if (data[0][0].status !== REGISTRATION_STATUS.UNREGISTERED){
+        return res.send({status : STATUS_CODE.USER_EXISTENT});
```
Comment (Vivian-liyw) [inline controllers/userController.js:288]: I think the log in APIs look great!
Quick question, what does user_existent stand for (lol)?

File: index.js (hunk 1 above)
Comment (KurtXiaoZ) [reply index.js:10]: Not sure...

File: controllers/userController.js [hunk 3]
```diff
@@ -1,90 +1,348 @@
...
+
+    let result = await this.insertUser({ email, name, type, visions });
+    if (result.affectedRows) {
+      return res.send({ status: STATUS_CODE.SUCCESS });
+    }
+  } catch (error){
+    return res.send({ status: STATUS_CODE.ERROR, result: error });
+  }
+
+  return res.send({ status: STATUS_CODE.ERROR });
```
Comment (Vivian-liyw) [inline controllers/userController.js:347]: I think we can reformat the code to have all the APIs for user log in and registration in another controller, but I think this is just a minor suggestion to consider if we have time later on

File: models/userManagement.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
+const { query } = require('./model');
```
Comment (Vivian-liyw) [inline models/userManagement.js:1]: I think this was from the old main branch, everything has been moved to userController. 
We can remove this entirely :)

File: controllers/userController.js (hunk 2 above)
Comment (KurtXiaoZ) [reply controllers/userController.js:288]: @OliviaLiu417 wrote this part. I think it means the user has already signed up so he or she can't sign up again

**Review by Vivian-liyw — COMMENTED** [review]
Great work, I think it looks amazing! We might need to delete the userMangament.js in models folder. But otherwise I think it's good to go!

File: controllers/userController.js (hunk 3 above)
Comment (KurtXiaoZ) [reply controllers/userController.js:347]: I agree!!!

File: index.js (hunk 1 above)
Comment (Vivian-liyw) [reply index.js:10]: I think we can delete this. I believe I wrote this previously for the middleware

File: controllers/userController.js (hunk 2 above)
Comment (OliviaLiu417) [reply controllers/userController.js:288]: Yes that means when a user has registered but tries to register again! I believe we should prevent this behavior

**Pull request merged**